### ISSUE TITLE
cudnnBatchNormMode_t mode_ not defined when compiling with CuDNN 3

### ISSUE THREAD

**Issue by mfernezir** [NVIDIA/caffe#85]
Hello,

I'm getting this error when compiling Caffe 0.14 with CuDNN 3. However, if i revert to the version '0.14.0-rc.1' which I downloaded a week ago then there is no error and everything compiles fine. There is a difference in common_layers.hpp, the earlier one doesn't mention cudnnBatchNormMode_t mode_ at all.

**Comment by thatguymike** [comment]
0.14 is being staged for cuDNNv4 and there is batch normalization support via cuDNN added to the tree.  I'll see if I can come up with a quick workaround to disable batchnorm support/deal with the API differences until cuDNNv4 is public.

**Comment by lukeyeager** [comment]
cuDNN v4 is now public ([website](https://developer.nvidia.com/cudnn), [packages](http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1404/x86_64/), [installation instructions](https://github.com/NVIDIA/DIGITS/blob/digits-3.0/docs/UbuntuInstall.md)). Please use the latest version of NVcaffe and cuDNN.
